common/src/unifyfs_client_rpcs.h

```diff
@@ -282,6 +283,105 @@ MERCURY_GEN_PROC(unifyfs_heartbeat_in_t,
 MERCURY_GEN_PROC(unifyfs_heartbeat_out_t, ((int32_t)(ret)))
 DECLARE_MARGO_RPC_HANDLER(unifyfs_heartbeat_rpc)
 
+/* unifyfs_node_local_extents_get_rpc (client => server)
+ *
+ * returns node local extents for a file
+ * given a global file id */
+MERCURY_GEN_STRUCT_PROC(unifyfs_index_t,
+                 ((uint32_t)(file_pos))
+                 ((uint32_t)(log_pos))
+                 ((uint32_t)(length))
+                 ((int32_t)(gfid)))
+/* List of extents used in the output of RPC call*/
+typedef struct extents_list {
+    unifyfs_index_t value;
+    struct extents_list* next;
+} *extents_list_t;
```

Comment:
hiding the fact that this is a pointer-to-struct in the typedef is confusing for later uses of extents_list_t. I suggest using extents_list_t as the typedef for the structure, and then using explicit pointer logic in the associated code.

Reply:
I fixed this check my comment below